buildengine/build_kotlin_test.go

```diff
@@ -1,30 +1,42 @@
 package buildengine
```

Comment:
refactored this test a bit to share logic with `build_go_test`; there are a few related changes in the kotlin test files to accommodate the refactor